lib/plugins/block_actions.js

```diff
@@ -2,19 +2,77 @@ const Vec3 = require('vec3').Vec3
 
 module.exports = inject
 
+const CARDINALS = {
+  north: new Vec3(0, 0, -1),
+  south: new Vec3(0, 0, 1),
+  west: new Vec3(-1, 0, 0),
+  east: new Vec3(1, 0, 0)
+}
+
 function inject (bot, { version }) {
   const { instruments, blocks } = require('minecraft-data')(version)
+
+  function parseChestMetadata (chestBlock) {
+    const chestTypes = ['single', 'right', 'left']
+    let data
+
+    if (bot.supportFeature('doesntHaveChestType')) {
+      data = {
+        facing: Object.keys(CARDINALS)[chestBlock.metadata - 2]
+      }
+    } else {
+      data = {
+        waterlogged: !(chestBlock.metadata & 1),
+        type: chestTypes[(chestBlock.metadata >> 1) % 3],
+        facing: Object.keys(CARDINALS)[Math.floor(chestBlock.metadata / 6)]
+      }
+    }
+    return data
+  }
+
+  function getChestType (chestBlock) { // Returns 'single', 'right' or 'left'
+    if (bot.supportFeature('doesntHaveChestType')) {
+      const facing = parseChestMetadata(chestBlock).facing
+
+      // We have to check if the adjacent blocks in the perpendicular cardinals are the same type
+      if (facing === 'north' || facing === 'south') {
+        if (bot.blockAt(chestBlock.position.plus(CARDINALS.west)).name === chestBlock.name) {
+          return facing === 'north' ? 'left' : 'right'
+        } else if (bot.blockAt(chestBlock.position.plus(CARDINALS.east)).name === chestBlock.name) {
+          return facing === 'north' ? 'right' : 'left'
+        }
+      } else if (facing === 'west' || facing === 'east') {
+        if (bot.blockAt(chestBlock.position.plus(CARDINALS.north)).name === chestBlock.name) {
+          return facing === 'west' ? 'right' : 'left'
+        } else if (bot.blockAt(chestBlock.position.plus(CARDINALS.south)).name === chestBlock.name) {
+          return facing === 'west' ? 'left' : 'right'
+        }
+      }
+
```

Comment:
Too over complicated. A double chest cannot be next to another chest.

Provided the above assumption, the facing won't be on the same axis as the chest so we can make a `facingMap` like so:
```js
const facingMap = {
  north: { east: 'left', west: 'right' },
  south: { east: 'right', west: 'left' },
  ...
}

for (const [cardinal, vec] of Object.entries(CARDINALS)) {
  if (bot.blockAt(chestBlock.position.plus(vec)).name === chestBlock.name) {
    return facingMap[cardinal][facing]
  }
}
```